pkg/action/executor/k8s_controller.go

```diff
@@ -136,14 +136,7 @@ func (c *parentController) update(updatedSpec *k8sControllerSpec) error {
 	}
 
 	controllerOwnerReferences := c.obj.GetOwnerReferences()
-	/*There are 2 conditions to determine whether to use ORM to update the workload controller or directly update it
-	1)the current workload contoller has the ownerReference and the owner has the controller field which is set to true
-	2)the current workload controller has the ownerReference and the kind of the owner is ClusterServiceVersion
-	*/
-	if !c.shouldSkipOperator(c.obj) && // Check if the label kubeturbo.io/skipOperator is set or not
-		len(controllerOwnerReferences) > 0 && // Check if the workcontroller has the owner
-		((controllerOwnerReferences[0].Controller != nil && *controllerOwnerReferences[0].Controller) || // case 1: owner exists and has a true controller field
-			"ClusterServiceVersion" == controllerOwnerReferences[0].Kind) { // case 2: owner exists and its kind is ClusterServiceVersion
+	if !c.shouldSkipOperator(c.obj) && c.shouldUseOrm(c.obj) {
```

Comment:
c.shouldSkipOperator does not check for c.obj being null, but c.shouldUseOrm does.  You may want to update shouldSkipOperator also

Reply:
Sure, I could add this check in the function `shouldSkipOperator `